File: 110-BalancedBinaryTree/110-BalancedBinaryTree.java
// Last updated: 2/27/2026, 10:35:42 PM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isBalanced(TreeNode root) {
        
        boolean ans[] = new boolean[1];
        ans[0] = true;

        traverse(root, ans);

        return ans[0];

    }

    public int traverse(TreeNode root, boolean ans[]){

        if(root == null) return 0;

        int l = traverse(root.left, ans);
        int r = traverse(root.right, ans);

        if( Math.abs(l - r) > 1) ans[0] = false;

        return Math.max(l,r) + 1;

    }
}

File: 110-BalancedBinaryTree/110-BalancedBinaryTree.py
# Last updated: 7/12/2026, 6:23:03 PM
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:

        def helper(node, isBalanced):
            if not node: return 0

            left = helper(node.left, isBalanced) + 1
            right = helper(node.right, isBalanced) + 1

            if abs(left - right) > 1: isBalanced[0] = False

            return max(left, right)

        isBalanced = [True]

        helper(root, isBalanced)

        return isBalanced[0]


        